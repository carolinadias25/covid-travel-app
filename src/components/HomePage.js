/* eslint-disable no-console */
/* eslint-disable spaced-comment */
/* eslint-disable quotes */
import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import "../styles/homepage.css";
import planepic from "../images/planepic.png";
import searchicon from "../images/search.png";
import countryList from "../data/countryNames";

export default function HomePage() {
  const [input, setInput] = useState("");
  const [error, setError] = useState(false);

  const history = useHistory();

  const search = (event) => {
    event.preventDefault();
    if (!countryList.includes(input)) {
      setError(true);
    } else {
      history.push(`/country-info/${input}`);
    }
  };

  return (
    <div className="container">
      <div className="search-section">
        <div className="search-section_small">
          <form className="search-box" onSubmit={search}>
            <h2 className="search-title">Welcome </h2>
            <img id="planepic" src={planepic} alt="" />
            <p className="search-question">Where are you headed?</p>
            <div className="blue-search-bar">
              <div className="white-search-bar">
                <input
                  value={input}
                  onChange={(e) => {
                    setInput(e.target.value);
                    setError(false);
                  }}
                  className="search-input"
                  type="text"
                  placeholder="e.g. Portugal"
                />
                <button className="search-button" type="submit">
                  <img id="search-icon" src={searchicon} alt="" />
                </button>
              </div>
            </div>
            {error && (
              <p className="homepage-error">
                Country does not exist - please check spelling
              </p>
            )}
          </form>
        </div>
      </div>
    </div>
  );
}
