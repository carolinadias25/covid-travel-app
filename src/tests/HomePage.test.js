/* eslint-disable */
import React from "react";
import { render } from "@testing-library/react";
import HomePage from "../components/HomePage";

describe("HomePage", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<HomePage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
