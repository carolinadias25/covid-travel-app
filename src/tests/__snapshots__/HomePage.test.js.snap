// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="search-section"
    >
      <div
        class="search-section_small"
      >
        <form
          class="search-box"
        >
          <h2
            class="search-title"
          >
            Welcome 
          </h2>
          <img
            alt=""
            id="planepic"
            src="planepic.png"
          />
          <p
            class="search-question"
          >
            Where are you headed?
          </p>
          <div
            class="blue-search-bar"
          >
            <div
              class="white-search-bar"
            >
              <input
                class="search-input"
                placeholder="e.g. Portugal"
                type="text"
                value=""
              />
              <button
                class="search-button"
                type="submit"
              >
                <img
                  alt=""
                  id="search-icon"
                  src="search.png"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
